### ISSUE TITLE
Proposal to form a Governance Working Group

### ISSUE THREAD

**Issue by dshaw** [nodejs/TSC#383]
Governance is a hard topic that requires research and iteration. It is political, people work, far from the technical work many of our collaborators are here for. The project has grown immensely over the last couple years and we are at an inflection point where we need to put in the time and effort to research solutions and bring a proposal before the TSC to be ratified.

At Collab Summit 2017, Sara Novotny shared how the Kubernetes project bootstrapped their governance process. One of the key details of that process was the Special Interest Group (SIG, analogous to our Working Groups) created by their Steering Committee, empowered and entrusted with creating the proposal. I think a similar process would benefit us at this time.

Proposed strategy:
* Invite a small team (3-5 people) who have the interests and skillset to represent us
    * Confirm that they have time
    * Ensure we have diverse representation amongst the individuals participating 
    * Seek out advisors to support these efforts if there are capabilities and/or representation gaps
* Empower them to work on our behalf
* Align on the acceptance criteria up-front
* Set expectations for timeline and sharing progress

**Comment by dshaw** [comment]
@mhdawson This is the "tsc-agenda" item [I suggested](https://github.com/nodejs/summit/issues/56#issuecomment-336516036). Process feedback and guidance appreciated.

**Comment by mcollina** [comment]
I am 👍  to this proposal. I would like to be part of the effort, but I will be sporadically available between now and December, and I cannot guarantee consistent output.

**Comment by jasnell** [comment]
I'm +1 on this and would definitely be willing to participate in the effort.

**Comment by mhdawson** [comment]
+1 and I'm willing to participate.

**Comment by geek** [comment]
Great idea. We should place limits on the number of representatives who can participate from a single company. Otherwise we risk having a governance proposal that favors the dominant participants.

**Comment by MylesBorins** [comment]
I have very mixed feelings about this. I think have a temporary interest group to come up with some definitive governance suggestions is a positive thing to do, but I am concerned that a working group could have the affect of removing governance responsibilities from the TSC. 

At Node Interactive we discussed putting together more specific guidelines around the expectations of a member of the TSC, and involvement in governance is one of those expectations.

I have been advocating for finding ways to further empower our collaborators to have more control over the technical management of the project, if we are able to be successful in this there should not be a need for individuals to be on the TSC for purely technical reasons.

**Comment by dshaw** [comment]
In the Kubernetes case, they chose to create a group that terminated its mandate at the completion of the initial governance work. We can choose to go that route. We also may want to consider a broader charter for our needs.

I held a role in High School student government called "Parliamentarian". The responsibility of this role was to be well versed in the established procedures and facilitate debate and decision making.

Care taking and assessing governance process and procedures does not necessarily equate to the actions required by project leaders to represent, administer, and decide.

**Comment by rvagg** [comment]
> Invite a small team (3-5 people) who have the interests and skillset to represent us

Who is "us" in here? What is the scope of "governance" in this context and how broad are the stakeholders that would make an "us"?

> Ensure we have diverse representation amongst the individuals participating

Depending on scope, "3-5 people" might be a challenge to get proper representation.

> Empower them to work on our behalf

I'd like clarity on what kind of empowerment is being proposed here. "empowered and entrusted with creating the proposal" sounds a bit fishy because nobody needs to be empowered or entrusted to make proposals, if you have a proposal then bring it! This seems like a pre-blessing on whatever they produce and I'd like to see clarity that it's not. Whatever the results of such a process it it's going to need broad buy-in to avoid further division.

**Comment by dshaw** [comment]
@rvagg The need for broad buy-in to the outcome and the heavy investment of time and effort are exactly why I am recommending that the TSC put in some effort up front to establish a committee of folks who can research, iterate and present a proposal for ratification.

**Comment by sarahnovotny** [comment]
I am also happy to participate in whatever way is useful to node.js.

**Comment by Fishrock123** [comment]
> In the Kubernetes case, they chose to create a group that terminated its mandate at the completion of the initial governance work. We can choose to go that route. We also may want to consider a broader charter for our needs.

Do keep in mind: **we are already past _initial_ governance work**, by two years at that.

**Comment by MylesBorins** [comment]
I would like to propose that we plan for a co-located in person multi-day meeting to discuss governance goals and steps to get us towards.

Any member of the committee who is unable to attend a colocation in person could attend digitally

Current suggested locations are Google NYC + Google Munich. We could theoretically also do something at Google in SF / Mountain View to cover the west coast.

**Comment by MylesBorins** [comment]
I would like to propose re-naming this effort as a `Node.js Governance Working Session`. It should include the entire TSC and invited observers. I apologize for the bike shedding on this but I think the framing here is very important. Framing it as a working session open to all TSC members, and a responsibility of all TSC members is extremely important to ensure that everyone can remain engaged.

**Comment by refack** [comment]
I was confused by the terminology as well. Who's the "us" who's the "them", and does this issue fall directly under the TSC privy? I keep thinking of the TSC as the __Technical__ steering committee, and this seems orthogonal to Technical.

IMHO having a SIG that is not comprised by TSC members, but rather by a "diverse representation amongst the individuals participating" is better way to go.

P.S. I would rather this thread be shelved, and moved to `nodejs/node` both for visibility, and because this repo has somewhat of a deterring effect on community members, and even Collaborators.

**Comment by MylesBorins** [comment]
TLDR; the job of the TSC is running the project, governance is a very important part of that

@refack if you look at the [tsc charter](https://github.com/nodejs/TSC/blob/master/TSC-Charter.md#section-5-responsibilities-of-the-tsc) you will see that the responsibilities of the TSC include:

> *  Setting release dates.
> *  Release quality standards.
> *  Technical direction.
> *  Project governance and process (including this policy).
> *  GitHub repository hosting.
> *  Conduct guidelines.
> *  Maintaining the list of additional Collaborators.
> *  Development process and any coding standards.
> *  Mediating technical conflicts between Collaborators or Foundation projects.

As you can see project governance + process is an expectation for the TSC. I personally believe that being engaged in developing the governance to be a key part of the role of a TSC member. While the TSC should definitely be talking to the general collaborators regarding decisions and coming up with process that empowers the entire project, this is very much something that should be designed and implemented by the TSC based on our current charter.

A sentiment among a number of members of the committee is that we would rather find ways to distribute the day to day technical conflict to allow the committee to focus on more forward thinking initiative. I'm going to open another thread about that right now 🎉

**Comment by sarahnovotny** [comment]
> Do keep in mind: we are already past initial governance work, by two years at that.

@Fishrock123 I've found that most projects need some effort spent every few years making sure that the governance they've outlined is actually what is happening. Where that isn't the case some introspection as to why processes and responsibilities have drifted.  The third piece of this effort is either a governance tuneup or a process/reality alignment with the documented ways.  Governance isn't static.

**Comment by refack** [comment]
>> Project governance and process (including this policy).
>
> As you can see project governance + process is an expectation for the TSC.

I stand corrected!
I would still hope to see inclusion of non-TSC members in this initiative ( @dshaw & @sarahnovotny are excellent examples ).

**Comment by jasnell** [comment]
I'd like the conversation to refocus back on the original proposal of organizing a small team to work out a proposal for updating the core project governance model. It is obvious that while the full TSC as it currently exists has an interest in the end result, only a subset of the TSC has any interest in being actively involved in the process. It is also obvious that whatever team does end up looking at this issue needs to include *more* than just current TSC members.

So what I'm going to do is this.

I ***am*** scheduling a call to talk specifically through TSC Governance. I've created a doodle poll here: https://doodle.com/poll/imwnmz98uad846p9 ... If any @nodejs/tsc or @nodejs/collaborators or @nodejs/foundation folks want to participate in the discussion, please fill out the doodle. If the times listed do not work at all, then we can add more options. I want to specifically invite @dhaw and @sarahnovotny to participate in those discussions. 

The purpose of this team discussion would be specifically:

1. Outline the specific issues with the current governance
2. Outline the specific *recommended* refactoring goals
3. Outline a *recommended* end vision
4. Outline the *recommended* refactoring steps

This team would *not* be long lived with an indefinite lifespan. It would be a short term effort designed to produce a very specific bit of output.

In regards to some of the specific pushback mentioned in the thread above:

@MylesBorins:
> but I am concerned that a working group could have the affect of removing governance responsibilities from the TSC.

That is not what is being proposed so this is not an issue.

@rvagg
> Depending on scope, "3-5 people" might be a challenge to get proper representation

Agreed. I don't think we should put a target number here. We should open the opportunity up to any collaborator or foundation representative who wants to participate and go with whatever number we actually get.

@rvagg 
> because nobody needs to be empowered or entrusted to make proposals, if you have a proposal then bring it!

This is fine in theory but really hasn't worked out this way in practice. In reality, proposals rarely make progress because the TSC is not actually engaged in the business of making progress. It's going to take an effort to build a workable proposal and those of who feel that it is necessary would like some reasonable assurance that the effort is not going to be wasted. To do so, we would like to have some sense that the TSC as a whole supports, if not encourages, the effort.

@MylesBorins 
> I would like to propose that we plan for a co-located in person multi-day meeting to discuss governance goals and steps to get us towards.

While getting everyone in the same location for a face-to-face is a noble idea, and likely worthwhile, it is not practically feasible within the near term. It also brings along it's own challenges for those who either (a) aren't invited or (b) can't participate for legitimate reasons. We should not hold this effort up waiting for the right opportunity to get people face to face.

**Comment by fhinkel** [comment]
I'm in favor of having a small team in charge instead.

**Comment by MylesBorins** [comment]
> That is not what is being proposed so this is not an issue.

From the original post

> Governance is a hard topic that requires research and iteration. It is political, people work, far from the technical work many of our collaborators are here for

That sentence in particular made it sound as thought it was framing the governance working group to offload political work from TSC. 

As it has become clear this is more of a breakout session for a small group to workshop proposals I see this as less of an issue. That being said I would like to see the TSC actively engaged in the discussions before and after the working sessions to ensure that all voices are being heard and there is equal ownership

**Comment by Fishrock123** [comment]
I've filled into the doodle - since those times are for this week we should probably pick something soon.

**Comment by jasnell** [comment]
I will pick a time later on today.

**Comment by MylesBorins** [comment]
@jasnell I think that we should consider pushing this to next week, at the moment it feels a bit rushed. Also, the timeslots are heavily skewed for the west coast, I'd like to see a slightly wider set of time options and see if that is a bit more flexible for those looking to attend

**Comment by dshaw** [comment]
@MylesBorins I intentionally have not edited my original post. If I should [update anything](https://github.com/nodejs/TSC/issues/383#issuecomment-339009086), I welcome your feedback on best practices since I'm unfamiliar with expectations here.

**Comment by jasnell** [comment]
Given that only 6 TSC members responded to the doodle, and given that it's already half way through the week, I'm going to reset the doodle with more times over the next few weeks and try this again. If we only get six TSC members participating, then fine, but that would be rather disappointing.

**Comment by dshaw** [comment]
@jasnell I missed the Doodle in the first round. I'll add my availability.

**Comment by mcollina** [comment]
I'm willing to partecipate if I am available.

**Comment by hackygolucky** [comment]
I think I'm a little confused just because there's been a lot of folks mentioned, but the scope of this is just TSC governance, yes?

**Comment by jasnell** [comment]
Yes, TSC governance.

All... I've expanded the doodle out to include more dates and times across the next few weeks, keeping in mind that several of us are going to be tied up at NodeConfEU in just over a week. If you have already weighed in, please update your availability.

ping @nodejs/tsc

**Comment by MylesBorins** [comment]
@jasnell would you be open to making a new doodle that has time zones? I have a fear that it isn't obvious that it is pst only and will lead to miscommunication

**Comment by dshaw** [comment]
Thank you, @jasnell. Updated my availability.

https://doodle.com/poll/imwnmz98uad846p9

**Comment by Fishrock123** [comment]
Just updated mine.

**Comment by MylesBorins** [comment]
@jasnell any word on a date for this?

**Comment by MylesBorins** [comment]
On monday I head to asia for two weeks making my availabiltiy a bit harder to lock down

I believe @mcollina is about to leave town as well.

@jasnell please let me know if you want me to pick up scheduling this meeting

**Comment by jasnell** [comment]
Go for it. My availability is quickly dwindling down to nothing as well

**Comment by Fishrock123** [comment]
Removing from the agenda as there is no progress to discuss.

**Comment by dshaw** [comment]
@MylesBorins Would you like to try to kick this offer before the end of the year?

**Comment by MylesBorins** [comment]
@dshaw in a previous TSC meeting it was decided that we didn't need to move forward with this in the immediate future. I think it makes sense to kick off an initial meeting in the new year when people are back from various vacations. Perhaps we can start a call for a date now?

**Comment by jasnell** [comment]
This appears to have stalled. Closing for now. Can reopen if any progress is made on moving forward on this.
